scripts/microshift-cleanup-data.sh

```diff
@@ -145,9 +157,11 @@ while [ $# -gt 0 ] ; do
 done
 
 # Verify valid option combination
-! ${FULL_CLEAN} && ! ${OVN_CLEAN}  && usage "Either --all or --ovn option must be specified"
-! ${FULL_CLEAN} &&   ${KEEP_IMAGE} && usage "The --keep-images option can only be used with --all"
-  ${FULL_CLEAN} &&   ${OVN_CLEAN}  && usage "The --all and --ovn options are mutually exclusive"
+! ${FULL_CLEAN} && ! ${OVN_CLEAN}  && ! ${CERT_CLEAN} && usage "Either --all, --ovn or --cert option must be specified"
```

Comment:
Some of these combinations seem redundant, but not "wrong". For example, combining `--all` with anything else or `--keep-images` and `--cert`. Can we be more permissive and allow those and only fail when the combinations are not valid?

Reply:
Maybe, there is a way to simplify the code by making positional assumptions about the arguments and checking their amount. Is this where the question is coming from? Should we try simplifying the code?

Reply:
It's more a usability question. If I say "delete everything and delete the certs" that doesn't feel like it should be an error. Think about someone trying to write an ansible role to run this command, and automatically build up the argument list based on variables set in the playbook. The template for that command line could be simple if they check individual flags that map to the command line options. If when the "delete all" flag is set they must not pass any other arguments then the template becomes more complicated.

For our testing, if we know that some of the options are redundant and supersede each other, I don't think we need additional test scenarios.

Reply:
What you're suggesting is a valid approach, but it's a change in the current behavior. I did not change the usage, which states that the options are mutually exclusive

```
Usage: microshift-cleanup-data.sh <--all [--keep-images] | --ovn | --cert>
```

If we want to change this behavior, I prefer to do it in a separate task. Should I create one and follow-up on this?

Reply:
Yes, let's leave that for another PR